### PR TITLE
Hive integration for date/time

### PR DESCRIPTION
**Pull request by CuriousVini**
**Summary**
Adding support for date/time for hive. Made changes to `ObjectDeserializer` and `ObjectSerializer` to support logicalTypes. Also made changes to `SchemaConverter` which is used to set hive schema for Table datasets. 

Build: https://builds.cask.co/browse/CDAP-DUT6636-6

### CODE REVIEW

File: cdap-explore/src/main/java/co/cask/cdap/hive/serde/ObjectSerializer.java [hunk 1]
```diff
@@ -238,4 +257,23 @@ private Object fromLazyObject(TypeInfo type, Object data) {
         return data.toString();
     }
   }
+
+  private Long toDays(Date obj) {
+    return obj == null ? null : obj.toLocalDate().toEpochDay();
```
Comment (albertshau) [inline cdap-explore/src/main/java/co/cask/cdap/hive/serde/ObjectSerializer.java]: if obj can be null, it should be annotated accordingly. Alternatively, in the serialize method, you can start off by returning null if the obj is null. That way you don't need this method and can just call obj.toLocalDate().toEpochDay() directly.

File: cdap-explore/src/main/java/co/cask/cdap/hive/serde/ObjectSerializer.java [hunk 2]
```diff
@@ -238,4 +257,23 @@ private Object fromLazyObject(TypeInfo type, Object data) {
...
+  }
+
+  private Object toEpochTimestamp(Timestamp ts, Schema schema) {
+    if (ts == null) {
```
Comment (albertshau) [inline cdap-explore/src/main/java/co/cask/cdap/hive/serde/ObjectSerializer.java]: similar comment, you can do the null check in the serialize method

**Review by albertshau — COMMENTED** [review]
minor comments.

**Review by albertshau — DISMISSED** [review]
lgtm

**Comment by CuriousVini** [comment]
The latest build is green. Squashing

**Review by albertshau — APPROVED** [review]
lgtm

**Pull request merged**